cli/module_build.go

```diff
@@ -490,6 +498,144 @@ type reloadModuleArgs struct {
+	return filepath.Join(stringDir, viamReloadArchive), nil
+}
+
+// moduleCloudReload triggers a cloud build and then reloads the specified module with that build.
+func (c *viamClient) moduleCloudReload(ctx *cli.Context, args reloadModuleArgs, platform string) (string, error) {
+	manifest, err := loadManifest(args.Module)
+	if err != nil {
+		return "", err
+	}
+
+	id := ctx.String(generalFlagID)
+	if id == "" {
+		id = manifest.ModuleID
+	}
+
+	archivePath, err := c.createGitArchive(args.Path)
+	if err != nil {
+		return "", err
+	}
+
+	moduleID, err := parseModuleID(manifest.ModuleID)
+	if err != nil {
+		return "", err
+	}
+	org, err := getOrgByModuleIDPrefix(c, moduleID.prefix)
+	if err != nil {
+		return "", err
+	}
+
+	// Upload a package with the bundled local dev code. Note that "reload" is a sentinel
+	// value for hot reloading modules. App expects it; don't change without making a
+	// complimentary update to the app repo
+	resp, err := c.uploadPackage(org.GetId(), reloadVersion, reloadVersion, "module", archivePath, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// get package URL for downloading purposes
+	packageURL, err := c.getPackageDownloadURL(org.GetId(), reloadVersion, reloadVersion, "module")
+	if err != nil {
+		return "", err
+	}
+
+	// object ID stringifies as `ObjectID("{actual_id}")` but we only want the actual ID when
+	// passing back to app for package lookup.
+	versionParts := strings.Split(resp.Version, "\"")
+	if len(versionParts) != 3 {
+		return "", errors.Errorf("malformed ID %s", versionParts)
+	}
+	resp.Version = versionParts[1]
+
+	// (TODO RSDK-11531) It'd be nice to add some streaming logs for this so we can see how the progress is going. create a new build
+	// TODO (RSDK-11692) - passing org ID in the ref field and `resp.Version` (which is actually an object ID)
+	// in the token field is pretty hacky, let's fix it up
+	infof(c.c.App.Writer, "Creating a new cloud build and swapping it onto the requested machine part. This may take a few minutes...")
+	buildArgs := moduleBuildStartArgs{
+		Module:    args.Module,
+		Version:   reloadVersion,
+		Workdir:   args.Workdir,
+		Ref:       org.GetId(),
+		Platforms: []string{platform},
+		Token:     resp.Version,
+	}
+
+	buildID, err := c.moduleBuildStartForRepo(ctx, buildArgs, &manifest, packageURL)
+	if err != nil {
+		return "", err
+	}
+
+	// ensure the build completes before we try to dowload and use it
+	_, err = c.waitForBuildToFinish(buildID, platform)
+	if err != nil {
+		return "", err
+	}
+
+	// (TODO RSDK-11517) There seems to be some delay between a build finishing and being findable
+	// for download. In testing, a delay of 3 seconds wasn't reliably long enough but 5 seconds was.
+	time.Sleep(time.Second * 5)
+
+	downloadArgs := downloadModuleFlags{
+		ID:       id,
+		Version:  reloadVersion,
+		Platform: platform,
+	}
+
+	// delete the package now that the build is complete
+	_, err = c.packageClient.DeletePackage(
```

Comment:
not a merge blocker given how this feature will be used, but this may be a race condition? like this may delete a different reload build

Reply:
I think it's a unique package based on the ID? We're specifying the package that we just uploaded, unless I'm missing something.